### PR TITLE
MM-49869 -  add channels Cta button to the LHS sidebar

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds a new CTA button to the LHS for users to add channels.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49869

#### Related Pull Requests
n/a

#### Screenshots
Before:
Screenshot 2023-03-06 at 17 20 28

After:
Screenshot 2023-03-06 at 17 21 28

https://user-images.githubusercontent.com/10082627/223169580-7b601183-9bac-4f43-a037-d6b38f929ec9.mov

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/sidebar/add_channels_cta_button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        expect(mockDispatch).toHaveBeenCalled();
+    });
+
+    test('should fire trackEvent to send telemetry when button is clicked', () => {
+        const wrapper = mountWithIntl(
+            <AddChannelsCtaButton/>,
+        );
+
+        const button = wrapper.find('.AddChannelsCtaDropdown button');
+        expect(mockDispatch).not.toHaveBeenCalled();
```
Comment (neallred) [inline components/sidebar/add_channels_cta_button.test.tsx:143]: Nit: `expect(trackEvent).not.toHaveBeenCalled();`

File: components/sidebar/add_channels_cta_button.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    category: Preferences.TOUCHED,
+                    user_id: currentUserId,
+                    name: Touched.ADD_CHANNELS_CTA,
+                    value: 'true',
+                }],
+            ));
+        }
+    };
+
+    if (!canCreateChannel && !canJoinPublicChannel) {
```
Comment (neallred) [inline components/sidebar/add_channels_cta_button.tsx:125]: Nit: Might make sense to group this early return next to the other one up above.

File: components/sidebar/invite_members_button.tsx [hunk 3]
```diff
@@ -50,10 +50,10 @@ const InviteMembersButton: React.FC<Props> = (props: Props): JSX.Element | null
         props.onClick();
     };
 
-    let buttonClass = 'SidebarChannelNavigator_inviteMembersLhsButton';
+    let buttonClass = 'SidebarChannelNavigator__inviteMembersLhsButton';
```
Comment (neallred) [inline components/sidebar/invite_members_button.tsx:53]: :+1:

File: utils/constants.tsx [hunk 4]
```diff
@@ -163,6 +163,7 @@ export const Preferences = {
 // For one off things that have a special, attention-grabbing UI until you interact with them
 export const Touched = {
     INVITE_MEMBERS: 'invite_members',
+    ADD_CHANNELS_CTA: 'add_channels_cta',
```
Comment (neallred) [inline utils/constants.tsx:166]: :tada: glad `Touched` is getting more use, seemed like overkill when I added it and it was all by itself for months.

File: sass/layout/_sidebar-left.scss [hunk 5]
```diff
@@ -178,6 +178,46 @@ $sidebarOpacityAnimationDuration: 0.15s;
             }
         }
 
+        &__inviteMembersLhsButton,
+        &__addChannelsCtaLhsButton {
```
Comment (neallred) [inline sass/layout/_sidebar-left.scss:182]: Nice refactor to keep this DRY.

**Review by neallred — APPROVED** [review]
LGTM, thanks @pvev ! A couple non-blocking nits.

File: components/sidebar/add_channels_cta_button.tsx (hunk 2 above)
Comment (julmondragon) [reply components/sidebar/add_channels_cta_button.tsx:125]: +1

**Review by julmondragon — APPROVED** [review]
Looks good @pvev !
Just some comments/nits and make sure to check the failed lint check.

**Review by stevemudie — APPROVED** [review]
This looks and works great @pvev!

File: components/sidebar/add_channel_dropdown.tsx [hunk 6]
```diff
@@ -189,14 +189,12 @@ const AddChannelDropdown = ({
...
-                        aria-label={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.dropdownAriaLabel', defaultMessage: 'Add Channel Dropdown'})}
-                    >
-                        <i className='icon-plus'/>
-                    </button>
-                </>
+                <button
+                    className={'AddChannelDropdown_dropdownButton'}
+                    aria-label={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.dropdownAriaLabel', defaultMessage: 'Add Channel Dropdown'})}
+                >
+                    <i className='icon-plus'/>
```
Comment (AshishDhama) [inline components/sidebar/add_channel_dropdown.tsx:196]: @pvev I know all the reviews are done, but can we use compass icons here

Comment (pvev) [reply components/sidebar/add_channel_dropdown.tsx:196]: sure, but do you mean these? https://mattermost.github.io/compass-icons/

Comment (AshishDhama) [reply components/sidebar/add_channel_dropdown.tsx:196]: Yeah these.

**Comment by pvev** [comment]
Migrated to https://github.com/mattermost/mattermost-server/pull/22701

**Pull request closed without merging**